File: Choonghun/개인 문제/3주차/프로그래머스 43162 - 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    graph = [[] for _ in range(n)]
    check = [0 for _ in range(n)]
    for i in range(n):
        for j in range(n):
            if i!=j and computers[i][j]:
                graph[i].append(j)
    
    for i in range(n):
        if not check[i]:
            answer += 1
            que = deque([i])
            while len(que) > 0:
                cur = que.popleft()
                for next in graph[cur]:
                    if not check[next]:
                        que.append(next)
                        check[next] = 1 
              
    return answer

File: Wonyoung/개인 문제/7주차/프로그래머스 43162 - 네트워크/네트워크.java
import java.util.*;

class Solution {
    public static boolean[] visited;
    public static int[][] graph;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];
        graph = computers;

        for(int i = 0; i < n; i++){
            if(!visited[i]){
                bfs(i, n);
                answer++;
            }
        }
        return answer;
    }
    
    public static void bfs(int start, int size){
        Deque<Integer> queue = new ArrayDeque<>();
        queue.offerLast(start);
        visited[start] = true;

        while(!queue.isEmpty()){
            int cur = queue.pollFirst();
            for(int i = 0; i < size; i++){
                if(i != cur && !visited[i] && graph[cur][i] > 0){
                    visited[i] = true;
                    queue.offerLast(i);
                }
            }
        }
    }
}